auraed/src/cells/cell_service/cells/cgroups/cgroup.rs

```diff
@@ -28,117 +28,201 @@
  *                                                                            *
 \* -------------------------------------------------------------------------- */
 
+use super::Result;
+use crate::cells::cell_service::cells::cgroups::CgroupsError;
 use crate::cells::cell_service::cells::{
     cgroups::{CpuController, CpusetController},
     CellName, CgroupSpec,
 };
-use cgroups_rs::{cgroup_builder::CgroupBuilder, hierarchies, Hierarchy};
+use libcgroups::common::{CgroupManager, ControllerOpt, DEFAULT_CGROUP_ROOT};
+use libcgroups::stats::Stats;
+use libcgroups::v2;
 use nix::unistd::Pid;
+use oci_spec::runtime::{LinuxCpuBuilder, LinuxResourcesBuilder};
 use std::path::PathBuf;
+use std::str::FromStr;
 
 /// This is used as the denominator for the CPU quota/period configuration.  This allows users to
 /// set the quota as if it was in the unit "µs/s" without worrying about also setting the period.
 const MICROSECONDS_PER_SECOND: u64 = 1000000;
 
 #[derive(Debug)]
 pub struct Cgroup {
-    non_leaf: cgroups_rs::Cgroup,
-    leaf: cgroups_rs::Cgroup,
+    cell_name: CellName,
 }
 
 impl Cgroup {
-    pub fn new(cell_name: CellName, spec: CgroupSpec) -> Self {
+    pub fn new(
+        cell_name: CellName,
+        spec: CgroupSpec,
+        nested_auraed_pid: Pid,
```

Comment:
A safer signature would take `nested_auraed: &NestedAuraed` and then have the `let pid = nested_auraed.pid()` in the function, preventing passing a `Pid` from a different process.